### PR TITLE
Completed TPM2 wrappers and wrapper example code

### PR DESCRIPTION
**Pull request by dgarske**
* Added new TPM2 wrapper functions, structures and example code.
* Progress with supporting parameter encryption/decryption using both session auth and object auth.
* Enhanced the `wolfTPM2_GetRCString` to include the enum name and for non-debug builds shows only the enum name.
* Changed `TPM2_GetNonce` to use current ctx and not use arg.
* Cleanup with the response parsing to reduce code / simplify.
* Cleanup of bad arg return code and error message display.
* Cleanup of line lengths / formatting.

### CODE REVIEW

File: wolftpm/tpm2_wrap.h [hunk 1]
```diff
@@ -25,15 +25,81 @@
...
+} WOLFTPM2_HANDLE;
+
+typedef struct WOLFTPM2_SESSION {
+    WOLFTPM2_HANDLE handle;
+    TPM2B_NONCE     nonceTPM;
+} WOLFTPM2_SESSION;
+
+typedef struct WOLFTPM2_KEY {
+    WOLFTPM2_HANDLE   handle;
+    TPM2B_PRIVATE     private;
```
Comment (JacobBarthelmeh) [inline wolftpm/tpm2_wrap.h]: private or public will cause problems if trying to build with g++

Comment (JacobBarthelmeh) [reply wolftpm/tpm2_wrap.h]: please also do a check through the for other locations of public/private

File: src/tpm2.c [hunk 2]
```diff
@@ -4686,7 +4442,7 @@ TPM_RC TPM2_NV_Certify(NV_Certify_In* in, NV_Certify_Out* out)
     TPM2_CTX* ctx = TPM2_GetActiveCtx();
 
     if (ctx == NULL || in == NULL || out == NULL || ctx->authCmd == NULL)
-        return TPM_RC_BAD_ARG;
+        return BAD_FUNC_ARG;
```
Comment (JacobBarthelmeh) [inline src/tpm2.c]: Why the change from separate error codes for wolfTPM to dependent on wolfSSL error codes?

File: wolftpm/tpm2_wrap.h (hunk 1 above)
Comment (dgarske) [reply wolftpm/tpm2_wrap.h]: Good find! Making fix for that.

File: src/tpm2.c (hunk 2 above)
Comment (dgarske) [reply src/tpm2.c]: The TPM_RC codes are from the TCG specification and adding non-standard codes for those isn't ideal. Since we have this strong dependency on wolfSSL I thought we'd just use wolf error codes. In fact the wolfTPM2_GetRCString function calls into wc_GetErrorString for negative values.
